openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/UserRepository.java

```diff
@@ -65,6 +65,15 @@ public UserRepository(CollectionDAO dao, SecretsManager secretsManager) {
     this.secretsManager = secretsManager;
   }
 
+  public final Fields getFieldsWithUserAuth(String fields) {
+    if (fields != null && fields.equals("*")) {
+      List<String> tempFields = getAllowedFieldsCopy();
```

Comment:
@mohityadav766, why are we adding this as additional temp fields and also we shouldn't be returning this if the logged-in user is requesting for other users, even for admin

Reply:
@harshach the problem is whenever we are using createOrUpdate we want to make sure to have all the userData first Fetched , and then do createOrUpdate, otherwise what happens is that the fields gets overwritten , let say I don't have "authnticationMechanism" it would be null on fetch of user , and createOrUpdate will update it to null !!